File: testdata/lang/structure.cs
class Structure {
    int Run(int value) {
        System.Func<int, int> adjust = step => {
            foreach (int item in new[] { 1, 2 }) {
                if (item > 0) {
                    value += item switch {
                        1 => step,
                        _ => 0,
                    };
                }
            }
            return value;
        };
        try {
            while (value < 4) value++;
        } catch (System.Exception) {
            return 0;
        } finally {
            value++;
        }
        return adjust(1);
    }
}

File: testdata/lang/structure.go
package structure

func run(value int) int {
	adjust := func(step int) int {
		for _, item := range []int{1, 2} {
			if item > 0 {
				switch item {
				case 1:
					value += step
				default:
					value++
				}
			}
		}
		return value
	}
	select {
	case <-make(chan int):
		return 0
	default:
		return adjust(1)
	}
}
